src/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/UnsafeNativeMethods.cs

```diff
@@ -0,0 +1,836 @@
+            public string StringVal;
+            [FieldOffset(8)]
+            public UInt32 Count;
+            [FieldOffset(12)]
+            public UInt32 Type;
+        };
+
+        [DllImport(WEVTAPI, CharSet = CharSet.Auto, SetLastError = true)]
+        [SecurityCritical]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool EvtFormatMessage(
+                             EventLogHandle publisherMetadataHandle,
+                             EventLogHandle eventHandle,
+                             uint messageId,
+                             int valueCount,
+                             EvtStringVariant[] values,
+                             [MarshalAs(UnmanagedType.I4)]EvtFormatMessageFlags flags,
+                             int bufferSize,
+                             [Out, MarshalAs(UnmanagedType.LPWStr)]StringBuilder buffer,
+                             out int bufferUsed
+                                        );
+
+        [DllImport(WEVTAPI, EntryPoint = "EvtFormatMessage", CallingConvention = CallingConvention.Winapi, CharSet = CharSet.Auto, SetLastError = true)]
+        [SecurityCritical]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool EvtFormatMessageBuffer(
+                             EventLogHandle publisherMetadataHandle,
+                             EventLogHandle eventHandle,
+                             uint messageId,
+                             int valueCount,
+                             IntPtr values,
+                             [MarshalAs(UnmanagedType.I4)]EvtFormatMessageFlags flags,
+                             int bufferSize,
+                             IntPtr buffer,
+                             out int bufferUsed
+                                        );
+
+        //SESSION
+        [DllImport(WEVTAPI, CharSet = CharSet.Auto, SetLastError = true)]
+        [SecurityCritical]
+        internal static extern EventLogHandle EvtOpenSession(
+                            [MarshalAs(UnmanagedType.I4)]EvtLoginClass loginClass,
+                            ref EvtRpcLogin login,
+                            int timeout,
+                            int flags
+                                        );
+
+        //BOOKMARK
+        [DllImport(WEVTAPI, EntryPoint = "EvtCreateBookmark", CharSet = CharSet.Auto, SetLastError = true)]
+        [SecurityCritical]
+        internal static extern EventLogHandle EvtCreateBookmark(
+                            [MarshalAs(UnmanagedType.LPWStr)] string bookmarkXml
+                                        );
+
+        [DllImport(WEVTAPI, CharSet = CharSet.Auto, SetLastError = true)]
+        [SecurityCritical]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool EvtUpdateBookmark(
+                            EventLogHandle bookmark,
+                            EventLogHandle eventHandle
+                                        );
```

Comment:
can you please follow interop guidelines for all these new funcitons ? https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md